### PR TITLE
Introduce interface for derived measures

### PR DESCRIPTION
**Pull request by xMort**
Create a new interface for derived measures that will allow us to work with them in a unified way. The pull request resolves issue https://github.com/gooddata/gooddata-java/issues/622.

### CODE REVIEW

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.executeafm.afm;
+
+/**
+ * The interface of the {@link MeasureDefinition} classes that are derived from the master measure and have the identifier of the master measure.
+ */
+public interface DerivedMeasureDefinition extends MeasureDefinition {
```
Comment (peter-plochan) [inline src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: Consider to make this as abstract class with `measureIdentifier` property

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
```
Comment (jimirocks) [inline src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: please use OpenSource copyright

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import static com.gooddata.util.Validate.notNull;
+
+/**
+ * The interface of the {@link MeasureDefinition} classes that are derived from the master measure and have the identifier of the master measure.
```
Comment (jimirocks) [inline src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: It's not an interface...

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = PopMeasureDefinition.class, name = PopMeasureDefinition.NAME),
+        @JsonSubTypes.Type(value = OverPeriodMeasureDefinition.class, name = OverPeriodMeasureDefinition.NAME),
+        @JsonSubTypes.Type(value = PreviousPeriodMeasureDefinition.class, name = PreviousPeriodMeasureDefinition.NAME)
+})
+public abstract class DerivedMeasureDefinition implements MeasureDefinition, Serializable {
+
+    private static final long serialVersionUID = -1203802872091017113L;
+
+    private final String measureIdentifier;
```
Comment (jimirocks) [inline src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: I wonder, whether it should be `protected`?

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java (hunk 2 above)
Comment (xMort) [reply src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: My mistake, it's fixed now.

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java (hunk 3 above)
Comment (xMort) [reply src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: Fixed.

File: src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java (hunk 4 above)
Comment (xMort) [reply src/main/java/com/gooddata/executeafm/afm/DerivedMeasureDefinition.java]: I made it protected and fixed its occurrences in the subclasses from getter to direct member access.

**Pull request merged**